### ISSUE TITLE
Franka Kinematics at Controller Level

### ISSUE THREAD

**Issue by jcarpinelli-bdai** [frankaemika/franka_ros2#63]
Today, the `FrankaRobotModel` class provides users forward kinematics utility functions which operate at the **hardware** level. This works when used in ROS 2 controllers which accept true Franka hardware, but does not work for simulated Franka arms. Is there some way to compute Franka kinematics using user-provided positions? I would be interested in helping to contribute this, if not.

**Comment by BarisYazici** [comment]
Yes currently for the gazebo simulator FrankaRobotModel and FrankaState interfaces are not available. However with Gazebo, you get joint_states(joint_positions, joint_velocities, joint_efforts). Based on these values you could calculate the M,C and G using either KDL or Pinocchio. For the Jacobians you could probably also use the same tools. Here is an example how we calculate the gravity term for the Gazebo simulation in KDL -> https://github.com/frankaemika/franka_ros2/blob/humble/franka_gazebo/franka_ign_ros2_control/src/model_kdl.cpp.
We will support FrankaRobotModel and FrankaState for Gazebo simulation, but the timeline is not clear. You are free to contribute. But currently the supervision from our side could be limited.

**Comment by dssdanial** [comment]
Hi, I just ran into the same issue when trying to test a controller in Gazebo. Is there any update on the plan to support FrankaRobotModel / FrankaState in simulation?
Thanks

**Comment by AndreasKuhner** [comment]
Hi @dssdanial ,
in general on the road map but unclear when we come to it. 

But stay focused and check the changelogs from time to time :) 

Cheers,
Andreas
